### ISSUE TITLE
Can't run RSpec system specs without Chrome installed, even specifying Rack::Test explicitly

### ISSUE THREAD

**Issue by dmolesUC** [rails/rails#37410]
RSpec tests configured to run with `Rack::Test` can't be run in a container (or on a machine) without Chrome, because `ActiveSupport::TestCase` tries to load Chrome as soon as it's required, resulting in `Webdrivers::BrowserNotFound`.

### Steps to reproduce

1. Clone the repository: [dmolesUC/rack_test_rails_6](https://github.com/dmolesUC/rack_test_rails_6), consisting of a minimal Rails 6 app with one RSpec-based system test, configured to run with `Rack::Test`.

2. Following the instructions in the README, stand up a Docker stack and attempt to run the specs.

   ```sh
   docker-compose up --pull && \
    docker-compose run --rm rails spec
   ```

### Expected behavior

Spec runs with Rack::Test.

### Actual behavior

Spec can't even be loaded, and fail with the following error:

```
Webdrivers::BrowserNotFound:
  Failed to find Chrome binary.
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chrome_finder.rb:21:in `location'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chrome_finder.rb:10:in `version'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chromedriver.rb:46:in `browser_version'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chromedriver.rb:106:in `release_version'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chromedriver.rb:32:in `latest_version'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/common.rb:135:in `correct_binary?'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/common.rb:91:in `update'
# /usr/local/bundle/ruby/2.6.0/gems/webdrivers-4.1.3/lib/webdrivers/chromedriver.rb:119:in `block in <main>'
# /usr/local/bundle/ruby/2.6.0/gems/activesupport-6.0.0/lib/active_support/core_ext/object/try.rb:15:in `public_send'
# /usr/local/bundle/ruby/2.6.0/gems/activesupport-6.0.0/lib/active_support/core_ext/object/try.rb:15:in `try'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_testing/browser.rb:50:in `preload'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_testing/driver.rb:13:in `initialize'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_test_case.rb:159:in `new'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_test_case.rb:159:in `driven_by'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_test_case.rb:162:in `<class:SystemTestCase>'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_test_case.rb:114:in `<module:ActionDispatch>'
# /usr/local/bundle/ruby/2.6.0/gems/actionpack-6.0.0/lib/action_dispatch/system_test_case.rb:16:in `<main>'
# /usr/local/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
# /usr/local/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
# /usr/local/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
# /usr/local/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
# /usr/local/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
# /usr/local/bundle/ruby/2.6.0/gems/zeitwerk-2.2.0/lib/zeitwerk/kernel.rb:23:in `require'
# /usr/local/bundle/ruby/2.6.0/gems/activesupport-6.0.0/lib/active_support/concern.rb:122:in `class_eval'
# /usr/local/bundle/ruby/2.6.0/gems/activesupport-6.0.0/lib/active_support/concern.rb:122:in `append_features'
# ./spec/system/foos_spec.rb:3:in `<top (required)>'
```

### System configuration
**Rails version**: 6.0.0

**Ruby version**: 2.6.4

**Comment by sinsoku** [comment]
There are two causes of this issue.

* System specs implicitly require 'action_dispatch/system_test_case' and driven_by(:selenium) is executed immediately in the file
* driven_by(:selenium) is always called in rspec-rails

I fixed driver configuration to be lazy for https://github.com/rails/rails/pull/37476 and https://github.com/rspec/rspec-rails/pull/2188.

**Comment by y-yagi** [comment]
Fixed by #37476.

**Comment by maysam** [comment]
I just wanted to mention here that I have similar problem, only that it works fine locally but chrome is required for rack_test when it is running on #circleci
